marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -172,17 +174,17 @@ getMarloweConstructors ContractType =
     [ (Tuple "Close" [])
     , (Tuple "Pay" [ DataArg "accountId", DataArg "payee", DataArg "token", DataArg "value", DataArg "contract" ])
     , (Tuple "If" [ DataArg "observation", DataArg "contract", DataArg "contract" ])
-    , (Tuple "When" [ ArrayArg "case", DataArg "timeout", DataArg "contract" ])
+    , (Tuple "When" [ ArrayArg "case", DefaultNumber zero, DataArg "contract" ])
     , (Tuple "Let" [ DataArg "valueId", DataArg "value", DataArg "contract" ])
     ]
 
-getMarloweConstructors BoundType = Map.singleton "Bound" [ DataArg "from", DataArg "to" ]
+getMarloweConstructors BoundType = Map.singleton "Bound" [ DefaultNumber zero, DefaultNumber zero ]
 
-getMarloweConstructors TokenType = Map.singleton "Token" [ DataArg "currency", DataArg "token" ]
+getMarloweConstructors TokenType = Map.singleton "Token" [ DefaultString "", DefaultString "" ]
 
 getMarloweConstructors PartyType =
   Map.fromFoldable
-    [ (Tuple "PK" [ DataArg "pubKey" ])
+    [ (Tuple "PK" [ DefaultString "pubKey" ])
     , (Tuple "Role" [ DataArg "token" ])
```

Comment:
This `Role` expansion is expanding to something that doesn't parse. Not sure were in the code the problem is

Reply:
I've now added tests that ensure that all `MarloweType` values are tested.